setup.cfg

```diff
@@ -11,7 +11,7 @@ classifiers =
   Development Status :: 5 - Production/Stable
 
 [options]
-python_requires = >=3.7
+python_requires = >=3.10
```

Comment:
someone jsut asked about 3.7 the other day; was this working before, or was just an incorrect designation that's now fixed?

Reply:
python <3.10 is not supported because of the `slots` argument to dataclass
python <3.8 is not supported by pre-commit>=3, which could easily be remedied by either removing pre-commit from requirements-dev or pinning a lower version

Reply:
@malkomes why are we using the `slots` argument of `dataclass`? Is this meant to be a performance improvement?

Reply:
yeah, just performance not really a must have

Reply:
We're also pinning versions of scipy and numpy that aren't compatible with python 3.7